Handle using the sync client in an async context for local tokenizers

abdullahkady opened the pull request:
Addressing this: https://github.com/langchain-ai/langchain-cohere/issues/19

harry-cohere commented:
We discussed and you'd like to try an alternative approach. Let me know when you'd like a review!

harry-cohere commented on src/cohere/manually_maintained/tokenizers.py:
```diff
@@ -12,39 +13,64 @@ def tokenizer_cache_key(model: str) -> str:
+
+    size = int(typing.cast(int, requests.head(tokenizer_url).headers.get("Content-Length")))
+    size_mb = round(size / 1024 / 1024, 2)
+    print(f"Downloading tokenizer for model {model}. Size is {size_mb} MBs.")
```
Did we decide on print versus logging? Logging is a better approach, but I can see that we don't do much of it within this SDK

abdullahkady replied:
I don't think we decided on anything. I saw that the SDK didn't have any logging setup, so I just went with print statements (since it's just progress indication for the user, not error/structured logs)

@billytrend-cohere do you have opinions on logging?

harry-cohere replied:
I would recommend choosing logging over print, as people who use this library will likely want some level of control over how and when log lines are printed.

harry-cohere reviewed (approved):
Looking good, thanks!

Would you be able to add integration tests?

abdullahkady commented:
> Looking good, thanks!
> 
> Would you be able to add integration tests?

@harry-cohere I added a test but it doesn't make a lot of sense, now that we have split the functions (just like the rest of the SDK). 
The test would need to be something like:
- **async** test (in terms of the test context itself)
- initialize **sync** client
- run all the sync client calls (defined in the `test_client.py` in the test's async context)

harry-cohere commented:
> > Looking good, thanks!
> > Would you be able to add integration tests?
> 
> @harry-cohere I added a test but it doesn't make a lot of sense, now that we have split the functions (just like the rest of the SDK). The test would need to be something like:
> 
> * **async** test (in terms of the test context itself)
> * initialize **sync** client
> * run all the sync client calls (defined in the `test_client.py` in the test's async context)

Perhaps just integration tests around the functionality, as it's intended to be used? Up to you though.
